x-pack/plugins/canvas/canvas_plugin_src/functions/browser/essql.ts

```diff
@@ -1,104 +0,0 @@
-/*
- * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
- * or more contributor license agreements. Licensed under the Elastic License
- * 2.0; you may not use this file except in compliance with the Elastic License
- * 2.0.
- */
-
-import {
-  ExpressionFunctionDefinition,
-  ExpressionValueFilter,
-} from '@kbn/expressions-plugin/common';
-import { lastValueFrom } from 'rxjs';
-import { searchService } from '../../../public/services';
-import { ESSQL_SEARCH_STRATEGY } from '../../../common/lib/constants';
-import { EssqlSearchStrategyRequest, EssqlSearchStrategyResponse } from '../../../types';
-import { getFunctionHelp } from '../../../i18n';
-
-interface Arguments {
-  query: string;
-  parameter: Array<string | number | boolean>;
-  count: number;
-  timezone: string;
-}
-
-export function essql(): ExpressionFunctionDefinition<
-  'essql',
-  ExpressionValueFilter,
-  Arguments,
-  any
-> {
-  const { help, args: argHelp } = getFunctionHelp().essql;
-
-  return {
-    name: 'essql',
-    type: 'datatable',
-    context: {
-      types: ['filter'],
-    },
-    help,
-    args: {
-      query: {
-        aliases: ['_', 'q'],
-        types: ['string'],
-        help: argHelp.query,
-      },
-      parameter: {
-        aliases: ['param'],
-        types: ['string', 'number', 'boolean'],
-        multi: true,
-        help: argHelp.parameter,
-      },
-      count: {
-        types: ['number'],
-        help: argHelp.count,
-        default: 1000,
-      },
-      timezone: {
-        aliases: ['tz'],
-        types: ['string'],
-        default: 'UTC',
-        help: argHelp.timezone,
-      },
-    },
-    fn: (input, args, handlers) => {
-      const search = searchService.getService().search;
-      const { parameter, ...restOfArgs } = args;
-      const req = {
-        ...restOfArgs,
-        params: parameter,
-        filter: input.and,
-      };
-
-      return lastValueFrom(
-        search.search<EssqlSearchStrategyRequest, EssqlSearchStrategyResponse>(req, {
-          strategy: ESSQL_SEARCH_STRATEGY,
```

Comment:
So canvas now uses a new essql function that uses a new generic strategy from data plugin? 
Compared to new generic strategy, canvas ESSQL_SEARCH_STRATEGY did internal pagination through scroll till the end server-side. 
If I understand correctly, then canvas now lost it. 

So we need to add it back somewhere.

Options I was thinking of:

1. Add it as an option to search strategy and do it internally server-side as it was before (bad because doesn't fit into async search and search polling idea until we support client-side search strategies)
2. Handle it on the expression function level, maybe fit it into the partial results framework

Reply:
what did the scroll achieve that we don't with the new strategy ? canvas on the client received the list of all results. It will get the same now ?